docs/guides/javascript.md

```diff
@@ -209,3 +209,44 @@ expect(result).toEqual([
   }),
 ]);
 ```
+
+### Using custom resolver
+
+Spectral lets you provide any custom $ref resolver. By default, http(s) and file protocols are resolved, relatively to the document Spectral lints against.
+If you'd like support any additional protocol or adjust the resolution, you are absolutely fine to do it.
+In order to achieve that, you need to create a custom json-ref-resolver instance.
+
+```js
+const path = require('path');
+const fs = require('fs');
+const { Spectral } = require('@stoplight/spectral');
+const { Resolver } = require('@stoplight/json-ref-resolver');
+
+const customFileResolver = new Resolver({
+  resolvers: {
+    file: {
+      resolve: ref => {
+        return new Promise((resolve, reject) => {
+          const basePath = process.cwd();
+          const refPath = ref.path();
```

Comment:
@P0lip I ran this locally, and this line looks a bit weird to me.

In the context of #1033, on my Win based box
- `ref.path()` returns `/spectral/paths/apis.yaml`
- `process.cwd()` equals `C:\spectral`
- `ref.toString()` returns `c:/spectral/paths/apis.yaml`.

The `path.join()` call below returns `C:\spectral\spectral\paths\apis.yml` which of course leads nowhere.

Could we have a doco bug here?

Reply:
> ref.path() returns /spectral/paths/apis.yaml

It's likely to be a urijs quirk :disappointed: 

> ref.toString() returns c:/spectral/paths/apis.yaml.

This is good provided that the file actually exists at given path. We should use `String(ref)` or `ref.toString()` in our code. The example is a bit wrong, we should update it.